client/state/partner-portal/products/reducer.ts

```diff
@@ -0,0 +1,33 @@
+import { AnyAction } from 'redux';
+import {
+	JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_ADD,
+	JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_REMOVE,
+	JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_CLEAR,
+} from 'calypso/state/action-types';
+import { combineReducers } from 'calypso/state/utils';
+import { ProductsStore } from '../types';
+
+const initialState: ProductsStore = {
+	selectedProductSlugs: [],
+};
+
+const selectedProductSlugs = ( state = initialState.selectedProductSlugs, action: AnyAction ) => {
+	switch ( action.type ) {
+		case JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_ADD:
+			return Array.isArray( action.productSlugs )
+				? [ ...state, ...action.productSlugs ]
+				: [ ...state, action.productSlugs ];
+		case JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_REMOVE:
+			return Array.isArray( action.productSlugs )
+				? state.filter( ( productSlug ) => ! action.productSlugs.includes( productSlug ) )
+				: state.filter( ( productSlug ) => productSlug !== action.productSlugs );
+		case JETPACK_PARTNER_PORTAL_SELECTED_PRODUCT_SLUGS_CLEAR:
+			return [];
+	}
+
+	return state;
+};
+
+export default combineReducers( {
+	selectedProductSlugs,
```

Comment:
This subtree originally had both `products` and `selectedProductSlugs`, and `disabledProductSlugs` and `totalSelectedCost` as selectors. These made up one distinct collection of state and derived state. 

Then with the feedback from P2 I updated things so that `products` would be accessed only through `react-redux`. I debated about putting it in one of the other subtrees. Either `licenses` or `partner` could work since this relates to the license purchase flow and a partner is the one selecting products, but conceptually I feel that products are their own entity. It just seems a bit strange now because the products proper are not being stored in Redux. If you feel strongly another way I'm happy to move them to another subtree though.

Reply:
No strong feelings! This was just a drive-by review with thoughts and suggestions and I don't intend to block the PR at all.